### PR TITLE
feat(video-conferencing-service): SFO-51 Session and archive APIs and unit tests

### PR DESCRIPTION
**Pull request by anitsingh-sf**
## Description

feat(video-conferencing-service): SFO-51 Session and archive APIs and unit tests

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql [hunk 1]
```diff
@@ -4,21 +4,23 @@ GRANT ALL ON SCHEMA videochatms TO public;
...
   created_on 				timestamptz DEFAULT current_timestamp NOT NULL ,
-  modified_on 				timestamptz ,
+  modified_on 			timestamptz ,
   is_deleted 				bool DEFAULT false NOT NULL ,
-  created_by 				integer ,
-  modified_by 				integer ,
+  created_by 				integer,
+  modified_by 			integer,
+  ext_id            varchar,
+  ext_metadata      json
```
Comment (vineet-suri) [inline services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql:23]: jsonb type

File: services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql [hunk 2]
```diff
@@ -27,23 +29,27 @@ CREATE TABLE videochatms.session_attendees
...
   session_id	 			varchar NOT NULL ,
-  attendee 					integer  ,
+  attendee 					varchar NOT NULL,
   created_on 				timestamptz DEFAULT current_timestamp NOT NULL ,
-  modified_on 				timestamptz ,
-  is_deleted 				bool DEFAULT false NOT NULL ,
+  modified_on 			timestamptz,
+  is_deleted 				bool DEFAULT false NOT NULL,
+  ext_id            varchar,
+  ext_metadata      json
```
Comment (vineet-suri) [inline services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql:37]: jsonb type

File: services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql [hunk 3]
```diff
@@ -27,23 +29,27 @@ CREATE TABLE videochatms.session_attendees
...
-	"before" 				json,
-	"after" 				json,
-    reference        		varchar,
+    acted_at  				timestamptz DEFAULT current_timestamp,
+    acted_entity     	varchar	NOT NULL,
+	  "before" 				  json,
+	  "after" 				  json,
+    reference        	varchar,
+    ext_id            varchar,
+    ext_metadata      json
```
Comment (vineet-suri) [inline services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql:53]: jsonb type

File: services/video-conferencing-service/src/controllers/video-chat-session.controller.ts [hunk 4]
```diff
@@ -59,14 +112,79 @@ export class VideoChatSessionController {
...
+      auditLogPayload.after = { errorMessage };
+      this.auditLogRepository.create(auditLogPayload);
+      throw new HttpErrors.BadRequest(errorMessage);
+    }
+
+    // check for schduled meeting:
+    if (session.isScheduled && session.scheduleTime) {
+      if (
+        moment()
+          .add(process.env.TIME_TO_START, 'minutes')
```
Comment (vineet-suri) [inline services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:172]: this parameter should be read from config, not env

File: services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql (hunk 1 above)
Comment (bhavyasf) [reply services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql]: done

File: services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql (hunk 2 above)
Comment (bhavyasf) [reply services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql]: done

File: services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql (hunk 3 above)
Comment (bhavyasf) [reply services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql]: done

File: services/video-conferencing-service/package.json [hunk 5]
```diff
@@ -25,7 +25,7 @@
     "pretest": "npm run clean && npm run build",
     "test": "echo \"No tests !\"",
```
Comment (vineet-suri) [inline services/video-conferencing-service/package.json:26]: Restore the test command.

Comment (bhavyasf) [reply services/video-conferencing-service/package.json:26]: check `test:dev`

**Pull request merged**